src/en/spyfakku/src/eu/kanade/tachiyomi/extension/en/spyfakku/SpyFakku.kt

```diff
@@ -95,116 +97,152 @@ class SpyFakku : HttpSource() {
     }
 
     override fun mangaDetailsRequest(manga: SManga): Request {
-        manga.url = Regex("^/archive/(\\d+)/.*").replace(manga.url) { "/g/${it.groupValues[1]}" }.replace("/__data.json", "")
-        return GET(baseApiUrl + manga.url, headers)
-    }
-
-    override fun pageListRequest(chapter: SChapter): Request {
-        chapter.url = Regex("^/archive/(\\d+)/.*").replace(chapter.url) { "/g/${it.groupValues[1]}" }.replace("/__data.json", "")
-        return GET(baseApiUrl + chapter.url, headers)
+        manga.url = Regex("^/archive/(\\d+)/.*").replace(manga.url) { "/g/${it.groupValues[1]}" }
+        return GET(baseUrl + manga.url.substringBefore("?") + "/__data.json", headers)
     }
 
     override fun getFilterList() = getFilters()
 
+    // Details
     private val dateReformat = SimpleDateFormat("EEEE, d MMM yyyy HH:mm (z)", Locale.ENGLISH)
     private val releasedAtFormat = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss", Locale.ENGLISH).apply {
         timeZone = TimeZone.getTimeZone("UTC")
     }
-    private val createdAtFormat = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss", Locale.ENGLISH).apply {
+    private val createdAtFormat = SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.ENGLISH).apply {
         timeZone = TimeZone.getTimeZone("UTC")
     }
 
+    private fun getAdditionals(data: List<JsonElement>): Triple<String?, Pair<String, String>, Long> {
+        val hentaiIndexes = json.decodeFromJsonElement<HentaiIndexes>(data[1])
+
+        val description = data[hentaiIndexes.description].jsonPrimitive.contentOrNull
+        val released_at = data[hentaiIndexes.released_at].jsonPrimitive.content
+        val created_at = data[hentaiIndexes.created_at].jsonPrimitive.content
+        val size = data[hentaiIndexes.size].jsonPrimitive.long
+
+        return Triple(description, Pair(released_at, created_at), size)
+    }
+    private fun <T> Collection<T>.emptyToNull(): Collection<T>? {
+        return this.ifEmpty { null }
+    }
+
     private fun Hentai.toSManga() = SManga.create().apply {
         title = this@toSManga.title
-        url = "/g/$id"
-        author = (circles?.emptyToNull() ?: artists)?.joinToString { it.name }
-        artist = artists?.joinToString { it.name }
-        genre = tags?.joinToString { it.name }
-        thumbnail_url = "$baseImageUrl/$hash/1/c"
+        url = "/g/$id?$pages&hash=$hash"
```

Comment:
changing url means users would need to migrate, maybe fetch it in the page list methods(?) since it appears to be used there

Reply:
Hello, sorry for the late reply. Does my recent change/commit align with what you're suggesting? Thank you.